### DISCUSSION TITLE
Should AddressType be string by default?

### DISCUSSION THREAD

**Discussion by vseehausen** [wagmi-dev/abitype#226] [Q&A] [answered]
Most of the time when I use Viem, I have issues with AddressType. 

* It does not recognize correct addresses that are imported from json
* Scrolling through the viem Discord, it seems like many people are casting their types manually.
* The check does not add value, as a correctly formatted address could still be incorrect for the given usecase. 

From my perspective the value add is low, the cost is high, and many people (me included) have to write more code to circumvent this check. 

Wouldn't it make more sense to use *string* as the default value for AddressType and let users that really need it [configure](https://abitype.dev/config) it to *`0x{string}`*?

**Comment by tmm** [comment, marked as answer]
At the moment, we still prefer it to be opt out instead of opt in. It's not that difficult to opt out — only 5 lines of code.

```ts
declare module 'abitype' {
  export interface Register {
    AddressType: string
  }
}
```

> It does not recognize correct addresses that are imported from json

Importing JSON files is an anti-pattern for strict TypeScript. If you are using plain JS, you can still do better by using a [`narrow(value)`](https://github.com/wevm/abitype/blob/main/packages/abitype/src/narrow.ts) function.

> Scrolling through the viem Discord, it seems like many people are casting their types manually.

I wouldn't consider this a representative example.

> The check does not add value, as a correctly formatted address could still be incorrect for the given usecase.

Strict types don't prevent all issues that might arise, like invalid runtime values, but they add a small amount of friction to avoid common mistakes. It also makes types more obvious in your IDE, e.g. autocomplete, hints, lint messages.

**Reply by vseehausen** [reply]
Ok, valid arguments. I will be silent :)

But still I think importing addresses from JSON is a real usecase for many crypto projects. Just think about a backend that interacts with many Uniswap pools. Or uses many tokens.
